docs/en/stack/ml/setup.asciidoc

```diff
@@ -5,100 +5,145 @@
-`machine_learning_user` built-in roles or the equivalent cluster privileges and 
-the following index privileges:
+If you use {ml} APIs, you must have the following cluster and index privileges:
 
 For full access:
 
+[%interactive]
+* [ ] `machine_learning_admin` built-in role or the equivalent cluster 
+privileges 
 * [ ] `read` and `view_index_metadata` on source indices
 * [ ] `read`, `manage`, and `index` on destination indices (for 
   {dfanalytics-jobs} only)
 
 For read-only access:
 
+[%interactive]
+* [ ] `machine_learning_user` built-in role or the equivalent cluster privileges
 * [ ] `read` index privileges on source indices
 * [ ] `read` index privileges on destination indices (for {dfanalytics-jobs}
-only)
+  only)
+
 
 [discrete]
-[[kib-security-privileges]]
-=== {kib} privileges
+[[kib-visibility-spaces]]
+=== {kib} feature visibility in Spaces
 
 In {kib}, the {ml-features} must be visible in your
-{kibana-ref}/xpack-spaces.html#spaces-control-feature-visibility[space] and your
-source {data-sources} must exist in the same space as your {ml} jobs.
+{kibana-ref}/xpack-spaces.html#spaces-control-feature-visibility[space]. To 
+control which features are visible in your space, use **{stack-manage-app}** > 
+_{kib}_ > _Spaces_.
+
+[role="screenshot"]
+image::spaces.jpg["Manage spaces in {kib}"]
 
-{kib} enables you to control access to the {ml-features} within each space. You 
-can manage your roles, privileges, and spaces in the **{stack-manage-app}** app 
-in {kib}. For more information, see 
-{ref}/security-privileges.html[Security privileges] and 
-{kibana-ref}/kibana-privileges.html[{kib} privileges].
+In addition to index privileges, source {data-sources} must also exist in the 
+same space as your {ml} jobs. These can be configured in **{stack-manage-app}** 
+> _{kib}_ > _{data-sources-caps}_.
 
-The `machine_learning_admin` and `machine_learning_user` roles grant access to 
-the {ml-features} in all {kib} spaces. Therefore, when you use {kib}, use custom 
-roles instead and set your {kib} privileges appropriately for each space.
 
-For full access to the {ml-features} in {kib}, you must have:
+Each {ml} job can be assigned to all, one or multiple spaces. This can be 
+configured in **{stack-manage-app}** > _Alerts and Insights_ > 
+_Machine Learning Jobs_. To assign a job to a space, select the spaces icon 
+shown in the job list.
+
+[role="screenshot"]
+image::assign-job-spaces.jpg["Assign machine learning jobs to spaces"]
+
+
+[discrete]
+[[kib-security-privileges]]
+=== {kib} user
+
+Within a {kib} space, for full access to the {ml-features}, you must have:
 
 [%interactive]
-* [ ] `all` {kib} privileges for the {ml-features} in the appropriate spaces
-* [ ] `read` and `view_index_metadata` index privileges on source indices
-* [ ] `read`, `manage`, and `index` index privileges on destination indices (for
-  {dfanalytics-jobs} only)
+* [ ] `All` {kib} privileges for the {ml-features}
```

Comment:
It has just struck me that we can be clearer here..

`All Kibana privileges for the machine learning features` -- the words "machine learning" are a long way from "all".

Should this be:
`Machine Learning: All` Kibana feature privileges.

This might be a nice way to differentiate Kibana and es privileges.

Reply:
I like this change - makes it easier to read and also easier to match the docs text with the options in the roles UI. We should apply this style to all Kibana feature privileges that we mention here (Machine Learning, Discover, Data View Management).

Reply:
Changed in 6796c74.